packages/mdc-textfield/mdc-textfield.scss

```diff
@@ -540,72 +607,87 @@ $mdc-textfield-disabled-border-on-dark: rgba(white, .3);
       }
     }
   }
-  // stylelint-enable plugin/selector-bem-pattern
-}
 
-.mdc-textfield--fullwidth {
-  display: block;
-  box-sizing: border-box;
-  width: 100%;
-  margin: 0;
-  border: none;
-  border-bottom: 1px solid $mdc-textfield-divider-on-light;
-  outline: none;
+  &:not(.mdc-textfield--upgraded) {
+    .mdc-textfield__input {
+      padding-top: $padding-inset;
+    }
+  }
 
-  // stylelint-disable plugin/selector-bem-pattern
+  .mdc-textfield__icon {
+    @include mdc-rtl-reflexive-position(right, 22px);
 
-  &:not(.mdc-textfield--multiline) {
-    height: 56px;
+    position: absolute;
+    top: 18px;
+    color: $mdc-textfield-underline-on-light-idle;
+    cursor: pointer;
+
+    @include mdc-theme-dark(".mdc-textfield") {
+      color: $mdc-textfield-dark-label;
+    }
   }
+}
 
-  &.mdc-textfield--multiline {
-    padding: 20px 0 0;
+.mdc-textfield--textarea-with-icon {
+  .mdc-textfield__input {
+    padding-right: 54px;
```

Comment:
This seems like it should be applied reflexively for RTL. Right now, if you switch to RTL and have long text in the textarea-with-icon example, you still have this large right padding, meanwhile the text can run into the icon on the left.

![image](https://user-images.githubusercontent.com/398379/30490878-5a55a9ce-9a09-11e7-9f83-46459d6bda8a.png)

Reply:
We're going to pull this out into its own branch. It needs more guidance from design. We can fix this then.